arch/arm/soc/nxp_imx/mcimx7_m4/soc.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <soc.h>
+#include "wdog_imx.h"
+#include "board.h"
+
+
+/* Initialize clock. */
+void SOC_ClockInit(void)
+{
+	/* OSC/PLL is already initialized by Cortex-A7 (u-boot) */
+
+	/*
+	 * Disable WDOG3
+	 *	Note : The WDOG clock Root is shared by all the 4 WDOGs,
+	 *	so Zephyr code should avoid closing it
+	 */
+	CCM_UpdateRoot(CCM, ccmRootWdog, ccmRootmuxWdogOsc24m, 0, 0);
+	CCM_EnableRoot(CCM, ccmRootWdog);
+	CCM_ControlGate(CCM, ccmCcgrGateWdog3, ccmClockNeededRun);
+
+	RDC_SetPdapAccess(RDC, BOARD_WDOG_RDC_PDAP, 3 << (BOARD_DOMAIN_ID * 2),
+		false, false); WDOG_DisablePowerdown(BOARD_WDOG_BASEADDR);
+
+	CCM_ControlGate(CCM, ccmCcgrGateWdog3, ccmClockNotNeeded);
+
+	/* We need system PLL Div2 to run M4 core */
+	CCM_ControlGate(CCM, ccmPllGateSys, ccmClockNeededRun);
+	CCM_ControlGate(CCM, ccmPllGateSysDiv2, ccmClockNeededRun);
+
+	/* Enable clock gate for IP bridge and IO mux */
+	CCM_ControlGate(CCM, ccmCcgrGateIpmux1, ccmClockNeededRun);
+	CCM_ControlGate(CCM, ccmCcgrGateIpmux2, ccmClockNeededRun);
+	CCM_ControlGate(CCM, ccmCcgrGateIpmux3, ccmClockNeededRun);
+	CCM_ControlGate(CCM, ccmCcgrGateIomux, ccmClockNeededRun);
+	CCM_ControlGate(CCM, ccmCcgrGateIomuxLpsr, ccmClockNeededRun);
+
+	/* Enable clock gate for RDC */
+	CCM_ControlGate(CCM, ccmCcgrGateRdc, ccmClockNeededRun);
+}
+
+void SOC_RdcInit(void)
+{
+	/* Move M4 core to specific RDC domain */
+	RDC_SetDomainID(RDC, rdcMdaM4, BOARD_DOMAIN_ID, false);
+}
+
+#ifdef CONFIG_UART_IMX
+static void nxp_mcimx7_uart_config(void)
+{
+
+#ifdef CONFIG_UART_IMX_UART_2
+	/* We need to grasp board uart exclusively */
+	RDC_SetPdapAccess(RDC, rdcPdapUart2, 3 << (BOARD_DOMAIN_ID * 2), false,
```

Comment:
need to cleanup magic number 3, probably a comment or something about why (BOARD_DOMAIN_ID * 2)